### PR TITLE
Add tooltip on Builder actions

### PR DESCRIPTION
**Pull request by matallo**
re: https://github.com/CartoDB/cartodb/issues/13095

- cartodb: https://github.com/CartoDB/cartodb/pull/13102
- deep-insights: https://github.com/CartoDB/deep-insights.js/pull/627
- cartoassets: https://github.com/CartoDB/CartoAssets/pull/178

---

**acceptance**

- check against "Add tooltip on Builder actions" spreadsheet all tooltips have been added

### CODE REVIEW

**Comment by matallo** [comment]
made some changes in https://github.com/CartoDB/cartodb/pull/13102/commits/b5249666f90a7fb5fa7942f026870926ef95cc8c to add tootlip to the legends title /cc @rubenmoya

File: lib/assets/core/javascripts/cartodb3/editor.js [hunk 1]
```diff
@@ -393,6 +394,7 @@ var editorTabPaneView = createEditorMenuTabPane([
     }
   }, {
     icon: 'settings',
+    tooltip: 'map-settings',
```
Comment (csobier) [inline lib/assets/core/javascripts/cartodb3/editor.js]: Small change: this should be "Map Options", to match the UI. https://docs.google.com/spreadsheets/d/13055ajt2PYPUsnCdOYCmFHKzFFcg0Zxz6Xfk4xqmD5k/edit#gid=0

**Comment by matallo** [comment]
pushed the first specs, I'm trying to accommodate them with the older tests, these are more truly unit oriented but it is quite a challenge

File: lib/assets/core/javascripts/cartodb3/dataset/dataset-view.js [hunk 2]
```diff
@@ -77,9 +77,7 @@ module.exports = CoreView.extend({
 
     var tooltip = new TipsyTooltipView({
       el: this.$('.js-editor-logo'),
-      title: function () {
-        return _t('back-to-dashboard');
-      },
+      title: _t('back-to-dashboard'),
```
Comment (rubenmoya) [inline lib/assets/core/javascripts/cartodb3/dataset/dataset-view.js]: I think it has to be a function:

screen shot 2017-11-28 at 17 26 19

I've tried that before and it didn't work

File: lib/assets/core/javascripts/cartodb3/editor.js (hunk 1 above)
Comment (rubenmoya) [reply lib/assets/core/javascripts/cartodb3/editor.js]: This is matching the name in the `en.json` file, if you check it here: https://github.com/CartoDB/cartodb/pull/13102/files#diff-75426b988ee7e728691a6ee313a35c78R5 you can see it says "Map options"

Comment (csobier) [reply lib/assets/core/javascripts/cartodb3/editor.js]: Thanks for confirming!

File: lib/assets/core/javascripts/cartodb3/editor/layers/analysis-views/default-layer-analysis-view.js [hunk 3]
```diff
@@ -49,21 +50,35 @@ module.exports = CoreView.extend({
...
     this.$el.toggleClass('has-error', status === 'failed');
 
     this.el.dataset.analysisNodeId = this.model.id;
 
+    this._initViews(title);
+
     return this;
   },
 
+  _initViews: function (title) {
```
Comment (rubenmoya) [inline lib/assets/core/javascripts/cartodb3/editor/layers/analysis-views/default-layer-analysis-view.js]: We're not using this argument or is a TODO?

**Comment by matallo** [comment]
I already pushed the tipsy specs @rubenmoya

File: lib/assets/core/javascripts/cartodb3/components/custom-carousel/custom-carousel-view.js [hunk 4]
```diff
@@ -48,17 +48,17 @@ module.exports = CoreView.extend({
   },
 
   _renderList: function () {
-    this.collection.each(function (mdl) {
-      this._createItem(mdl);
+    this.collection.each(function (model) {
```
Comment (rjimenezda) [inline lib/assets/core/javascripts/cartodb3/components/custom-carousel/custom-carousel-view.js]: good scouts 👍

**Comment by rubenmoya** [comment]
Deploying to `ded12`

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/list/list-item.js [hunk 5]
```diff
@@ -128,6 +137,23 @@ Backbone.Form.editors.List.Item = Backbone.Form.editors.Base.extend({
...
+  },
+
+  remove: function () {
+    this._removeTooltip();
+    this.editor.remove();
+
+    Backbone.Form.editors.Base.prototype.remove.apply(this);
+  },
+
+  clean: function () {
```
Comment (matallo) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/list/list-item.js]: do we really need this? I bet this is already taken care of when doing the `Backbone.Form.editors.Base.prototype.remove.apply(this);` we might be missing something in the clean here

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/select/select-view.js [hunk 6]
```diff
@@ -248,5 +274,9 @@ Backbone.Form.editors.Select = Backbone.Form.editors.Base.extend({
     if (!visible) {
       this._getButton().blur();
     }
+  },
+
+  clean: function () {
```
Comment (matallo) [inline lib/assets/core/javascripts/cartodb3/components/form-components/editors/select/select-view.js]: same than previously, doesn't the base editor take care of this?

File: lib/assets/core/javascripts/cartodb3/components/form-components/editors/list/list-item.js (hunk 5 above)
Comment (rubenmoya) [reply lib/assets/core/javascripts/cartodb3/components/form-components/editors/list/list-item.js]: Without it, it fails when testing "It should have no leaks"

Comment (matallo) [reply lib/assets/core/javascripts/cartodb3/components/form-components/editors/list/list-item.js]: oooh, I'll 🤐 then, thanks for clarifying!

**Comment by matallo** [comment]
I can't approve the PR because I opened it, but LGTM! 👍

**Pull request merged**